app/Enums/ProjectStage.php

```
@@ -20,7 +22,32 @@ public static function flow(): array
+        $flowValues = self::values();
+        $oldIndex = array_search($oldStage, $flowValues, true);
+        $newIndex = array_search($newStage, $flowValues, true);
+        if ($oldIndex === false || $newIndex === false) {
+            throw ValidationException::withMessages([
+                'stage' => 'Invalid stage provided.',
+            ]);
+        }
+        if ($newIndex < $oldIndex) {
+            throw ValidationException::withMessages([
+                'stage' => 'Cannot move to an earlier stage.',
+            ]);
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Validation\ValidationException' in method 'validateTransition'. (Clean Code Rules)

(StaticAccess)